modules/fbs/includes/fbs.loan.inc

```diff
@@ -55,6 +55,32 @@ function fbs_loan_list($account, $reset = FALSE) {
           'materials_number' => $loan->loanDetails->materialItemNumber,
         );
 
+        // Try to provide some meaningful renewal feedback.
+        if (empty($loan->isRenewable)) {
+          $renewed_ids = array();
+          if (module_exists('ding_session_cache')) {
+            $renewed_ids = ding_session_cache_get('fbs', 'renewed_ids', array());
+          }
+
+          // We maintain a list of renewed loans in this session, and if this
+          // loan is here we will give it the renewed status for the rest of the
+          // session.
```

Comment:
Why do we want to set which ids was renewed specifically for the FBS provider? I would think this approach would work across providers.

Reply:
I was trying to solve a specific problem in FBS-provider, but you're right; this could also be useful in other providers. I'll try to move it up in the layers.